napari/utils/progress.py

```diff
@@ -0,0 +1,138 @@
+import inspect
+from typing import Iterable, Optional
+
+from tqdm import tqdm
```

Comment:
Has this been added to our dependencies, and do we have sufficient min version? I had to upgrade from `tqdm-4.42.1` to `tqdm-4.60.0` for one of the examples to work

Reply:
Good catch, thanks @sofroniewn. Also yes, we need >=4.56.0 to support passing the `gui` parameter the way we currently are.

Reply:
Rather than making it a requirement we can do something similar to what @tlambert03's done in [magicgui](https://github.com/napari/magicgui/blob/f883590ecb307661ae00defed0cb8ce6576168b5/magicgui/tqdm.py#L8), where we try to import but raise an error if we can't find it - if people aren't using progress bars maybe we shouldn't make them install `tqdm`?

Reply:
Ok we should probably add that explicitly now. Curious how @tlambert03 and @jni feel about adding that dependency?

Reply:
I'm +1 on the dependency

Reply:
I've added the dependency in now

Reply:
Do you know why the tests didn't fail without tqdm? Was it already a test dependency? I would have thought they should have?

Reply:
I honestly have no idea. We did have one `test_add_layer` that was consistently failing the `min_req` action. 

>Was it already a test dependency

Hmm it's in there for release, so that's probably why. Should I update that to `>=4.56.0` as well?

Reply:
> Hmm it's in there for release, so that's probably why. Should I update that to >=4.56.0 as well?

Ah I see, no I think you can remove it entirely, doesn't need to be in both